worlds/messenger/client_setup.py

```diff
@@ -0,0 +1,167 @@
+        assets = request_data(MOD_URL)["assets"]
+        if len(assets) == 1:
+            release_url = assets[0]["browser_download_url"]
+        else:
+            for asset in assets:
+                if "TheMessengerRandomizerAP" in asset["name"]:
+                    release_url = asset["browser_download_url"]
+                    break
+            else:
+                messagebox("Failure", "Failed to find latest mod download", True)
+                raise RuntimeError("Failed to install Mod")
+
+        mod_folder = os.path.join(folder, "Mods")
+        os.makedirs(mod_folder, exist_ok=True)
+        with urllib.request.urlopen(release_url) as download:
+            with ZipFile(io.BytesIO(download.read()), "r") as zf:
+                for member in zf.infolist():
+                    zf.extract(member, path=mod_folder)
+
+        messagebox("Success!", "Latest mod successfully installed!")
+
+    def available_mod_update(latest_version: str) -> bool:
+        """Check if there's an available update"""
+        latest_version = latest_version.lstrip("v")
+        toml_path = os.path.join(folder, "Mods", "TheMessengerRandomizerAP", "courier.toml")
+        with open(toml_path, "r") as f:
+            installed_version = f.read().splitlines()[1].strip("version = \"")
+
+        logging.info(f"Installed version: {installed_version}. Latest version: {latest_version}")
+        # one of the alpha builds
+        return not latest_version.isnumeric() or tuplize_version(latest_version) > tuplize_version(installed_version)
+
+    from . import MessengerWorld
+    folder = os.path.dirname(MessengerWorld.settings.game_path)
+    if not courier_installed():
+        should_install = askyesnocancel("Install Courier",
+                                        "No Courier installation detected. Would you like to install now?")
+        if not should_install:
+            return
+        logging.info("Installing Courier")
+        install_courier()
+    if not mod_installed():
+        should_install = askyesnocancel("Install Mod",
+                                        "No randomizer mod detected. Would you like to install now?")
+        if not should_install:
+            return
+        logging.info("Installing Mod")
+        install_mod()
+    else:
+        latest = request_data(MOD_URL)["tag_name"]
+        if available_mod_update(latest):
+            should_update = askyesnocancel("Update Mod",
+                                           f"New mod version detected. Would you like to update to {latest} now?")
+            if should_update:
+                logging.info("Updating mod")
+                install_mod()
+            elif should_update is None:
+                return
+    if is_linux:
+        if url:
+            open_file(f"steam://rungameid/764790//{url}/")
+        else:
+            open_file("steam://rungameid/764790")
+    else:
+        os.chdir(Path(MessengerWorld.settings.game_path).parent)
```

Comment:
You never set back the working directory to Archipelago's root directory, which means that everything from this point forward uses the game's directory as the base path. This breaks everything else in the launcher.

Examples:

Launching another game client
![image](https://github.com/ArchipelagoMW/Archipelago/assets/35039/27e5b398-441c-4d22-a0da-e5238b0e9817)

Logs ending up in the game directory
![image](https://github.com/ArchipelagoMW/Archipelago/assets/35039/c9baeceb-4020-420b-8347-0b827b0319b3)

Reply:
whoops good catch.